src/ore/src/id_gen.rs

```diff
@@ -97,8 +108,155 @@ where
     }
 }
 
+/// A clone-able owned reference to an ID.
+///
+/// Once all of the [`IdHandle`]s referencing an ID have been dropped, we will then free the ID
+/// for later re-use.
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd)]
+pub enum IdHandle<T: From<u8> + AddAssign + PartialOrd + Copy> {
```

Comment:
I don't see where From<u8> or AddAssign are required. Are they needed or do I miss something?

Reply:
Unfortunately these are required on the enum definition because of the custom `Drop` impl. `fn free(...)` is only implemented on `IdAllocator` when `T` has these bounds. So we need the `Drop` impl to require these bounds for `T`. But Rust doesn't allow `Drop` to be implemented for only _some_ instances of a type, which would be the case if we required these bounds on `Drop`, but not on `IdHandle` itself.

tl;dr we need to carry these bounds for the entire type so we can use them in `Drop`.